integrated_channels/blackboard/views.py

```diff
@@ -0,0 +1,128 @@
+
+            HTTP 400 if code/state is not provided
+
+            HTTP 404 if state is not valid or contained in the set of registered enterprises
+
+    """
+    def get(self, request, *args, **kwargs):  # pylint: disable=arguments-differ
+        app_config = apps.get_app_config('blackboard')
+        oauth_token_path = app_config.oauth_token_auth_path
+
+        # Check if encountered an error when generating the oauth code.
+        request_error = request.GET.get('error')
+        if request_error:
+            raise APIException(
+                "Blackboard OAuth API encountered an error when generating client code - "
+                "error: {} description: {}".format(
+                    request_error,
+                    request.GET.get('error_description')
+                )
+            )
+
+        # Retrieve the newly generated code and state (Enterprise user's ID)
+        client_code = request.GET.get('code')
+        enterprise_customer_uuid = request.GET.get('state')
+        if not enterprise_customer_uuid:
+            raise ParseError("Enterprise ID (as 'state' url param) needed to obtain refresh token")
+
+        if not client_code:
+            raise ParseError("'code' url param was not provided, needed to obtain refresh token")
+
+        enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+        if not enterprise_customer:
+            raise NotFound("No enterprise data found for given uuid: {}."
+                           .format(enterprise_customer_uuid))
+
+        try:
+            enterprise_config = BlackboardEnterpriseCustomerConfiguration.objects.get(
+                enterprise_customer=enterprise_customer
+            )
+        except BlackboardEnterpriseCustomerConfiguration.DoesNotExist:
+            raise NotFound("No Blackboard configuration found for enterprise: {}"
+                           .format(enterprise_customer_uuid))
+
+        auth_header = self._create_auth_header(enterprise_config)
+
+        access_token_request_params = {
+            'grant_type': 'authorization_code',
+            'redirect_uri': settings.LMS_INTERNAL_ROOT_URL + "/blackboard/oauth-complete",
+            'code': client_code,
+        }
+
+        auth_token_url = urljoin(enterprise_config.blackboard_base_url, oauth_token_path)
+        auth_response = requests.post(
+            auth_token_url,
+            access_token_request_params,
+            headers={
+                'Authorization': auth_header,
+                'Content-Type': 'application/x-www-form-urlencoded'
+            })
+
+        try:
+            data = auth_response.json()
+            refresh_token = data['refresh_token']
+        except (KeyError, ValueError):
+            raise requests.RequestException(response=auth_response)
+
+        enterprise_config.refresh_token = refresh_token
+        enterprise_config.save()
+
+        return Response()
+
+    def _create_auth_header(self, enterprise_config):
```

Comment:
Seems like a very similar method to the one you have in the client, without the configuration check. Do you think it's worth making a common "create auth header" util? Not sure best practice here since it's such a minor function and only located in the two places (though perhaps larger use cases for integrated channels?)

Reply:
yeah I had thought of doing that, left it in place right now only due to it's smallness